terraform/modules/single_instance/cron.tf

```diff
@@ -0,0 +1,71 @@
+variable "crontabs" {
+  type = map(object({
+    cron_time = string
+    cron_user = optional(string,"root")
+    script = string 
+  }))
+  default = {}
+  validation {
+    condition = !( length(var.crontabs) > 0 && var.is_windows )
+    error_message = "Can't add crons to windows!"
+  }
+}
+
+resource "null_resource" "cron" {
+  for_each = {
+    for k,v in local.crons : k => v
+    if local.scripts_enabled
+  }
+  depends_on = [ null_resource.testconnection ]
+  triggers = {
+    user = local.ssh_user
+    port = local.ports.ssh
+    ip = data.openstack_networking_floatingip_v2.public.address
+    content = each.value.script
+  }
+  connection {
+    type     = "ssh"
+    user     = self.triggers.user
+    agent = true
+    host     = self.triggers.ip
+    timeout = "5m"
+    port = self.triggers.port
+  }
+  provisioner "file" {
+    destination = "/home/${local.ssh_user}/${random_id.obscure[each.key].id}-${each.key}.sh"
+    content = each.value.script
+  }
+  provisioner "file" {
+    destination = "/home/${local.ssh_user}/${random_id.obscure[each.key].id}-${each.key}.cron"
+    content = <<-EOT
+    ${each.value.cron_time} ${each.value.cron_user} /opt/vsc/cron/${each.key}.sh
+
+    EOT
+  }
+  provisioner "remote-exec" {
+    inline = [
+      "set -e",
+      "sudo cp /home/${local.ssh_user}/${random_id.obscure[each.key].id}-${each.key}.sh /opt/vsc/cron/${each.key}.sh",
```

Comment:
are these files only accessible by root user?

Reply:
The files are placed inside the user home directory first with default permissions, and are then immediately copied and chown-ed to root:

`"sudo chown root:root /etc/cron.d/${each.key}",`

There is a brief window where these files are not owned by root, but considering this user has access to sudo anyway that doesn't really make a difference. Nevertheless with an ansible solution we could place the files in the right directory immediately